### DISCUSSION TITLE
Why do few panels implement wayland taskbar preview.

### DISCUSSION THREAD

**Discussion by ffs-akl** [LBCrion/sfwbar#253] [General]
Of the panels I've come across, only plasma implements wayland taskbar preview.   Isn't it difficult to implement taskbar preview?

**Comment by Consolatis** [comment]
It needs a window capture / screenshot protocol which is not yet implemented in the wlroots based compositors.

**Reply by ffs-akl** [reply]
Oh,I see. Thank you for your reply.
So far, no the wlroots based compositors seem to be planning to implement it.

**Reply by Consolatis** [reply]
There is an [implementation](https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4545) for wlroots of the protocol which allows for output, cursor and window capture. However, from what I remember only the output and cursor capture parts are part of that MR (although I didn't follow its development for a month or so).

So chances are that it might be supported at some point but we aren't there yet.

**Reply by ffs-akl** [reply]
good to know,  It's a long way off.

**Comment by ffs-akl** [comment]
I noticed an exciting feature—window previews—has been implemented in commit 6eb3aa1.  I'd love to be one of the first to try it out! How can I enable it?

**Comment by LBCrion** [comment]
This feature requires a compositor that supports ext-toplevel-copy-capture
protocol,  ext-foreign-toplevel-list protocol and an ipc that allows
mapping windows to ext-foreign-toplevel-list stable ids.

Currently this realistically means the latest release candidate of sway.
